extensions/ftp/src/main/java/org/mule/extension/ftp/internal/sftp/connection/SftpClient.java

```diff
@@ -197,23 +197,23 @@ private void configureProxy(Session session) {
       switch (proxyConfig.getProtocol()) {
         case HTTP:
           ProxyHTTP proxyHttp = new ProxyHTTP(proxyConfig.getHost(), proxyConfig.getPort());
-          if (proxyConfig.getUsername() != null) {
+          if (proxyConfig.getUsername() != null && proxyConfig.getPassword() != null) {
```

Comment:
if one if provided without the other then it should fail instead of ignoring all credentials.